lib/Document/Adapter/Ghostscript.php

```diff
@@ -210,14 +210,22 @@ public function saveImage(string $imageTargetPath, $page = 1, $resolution = 200)
     /**
      * {@inheritdoc}
      */
-    public function getText(?int $page = null, ?Asset\Document $asset = null)
+    public function getText(?int $page = null, ?Asset\Document $asset = null, ?string $path = null)
     {
         try {
             if (!$asset && $this->asset) {
                 $asset = $this->asset;
             }
 
-            $path = $asset->getLocalFile();
+            if(!$path || !file_exists($path)) {
+                if(self::isFileTypeSupported($asset->getFilename())) {
+                    $path = $asset->getLocalFile();
+                }
+
+                if(empty($path)) {
+                    throw new \Exception('Variable $path can\'t be null at this point.');
```

Comment:
```suggestion
                    throw new \Exception('Could not get local file for asset with id ' . $asset->getId());
```
What do you think about adding a bit more detailed error message here?